src/27F70.c

```diff
@@ -0,0 +1,1743 @@
+#endif
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80031E38.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800320F8.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800327B4.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80032900.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80032E60.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800330A4.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033204.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800333A0.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033428.s")
+
+void func_8003358C(uint16_t index){
+    if(gActors[index].rgba.a<(int32_t)gActors[index].unk_0x18C_w) gActors[index].flag=0;
+    else gActors[index].rgba.a-=gActors[index].unk_0x18C_w;
+}
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800335E4.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800336B8.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800337F4.s")
+
+void func_800338F4(int16_t arg0, int16_t arg1, int16_t arg2) {
+    if ((gSceneFrames & 0x0F) == 0) {
+        func_800337F4(arg0, arg1, arg2, 0x132);
+    }
+}
+
+uint16_t D_800D2294[4] = { 0x136, 0x138, 0x13A, 0x13A };
+//spawn notes while idling
+void ActorMarina_Idle_SpawnNotes(int16_t arg0, int16_t arg1, int16_t arg2) {
+    if ((gSceneFrames & 0x0F) == 0) {
+        func_800337F4(arg0, arg1, arg2, D_800D2294[Rand() & 3]);
+    }
+}
+
+void func_800339AC(int16_t arg0, int16_t arg1, int16_t arg2) {}
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800339BC.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033B54.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033CB0.s")
+/* todo, set struct feilds to s2_w
+void func_80033DE4(uint16_t index){
+    if(--gActors[index].unk_0x154._w == 0) gActors[index].flag=0;
+    gActors[index].vel.x_w+=gActors[index].unk_0x158;
+    gActors[index].vel.y_w+=gActors[index].unk_0x15C;
+    gActors[index].unk_0x150._w = func_8002B5A0(gActors[index].unk_0x150._lo,gActors[index].unk_0x168._lo,gActors[index].unk_0x164._lo,gActors[index].unk_0x160._lo);
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033DE4.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033E7C.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800340CC.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80034644.s")
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_8003480C.s")
+
+/* How do I const floats?
+void func_800348E4(uint16_t index){
+    gActors[index].rgba.a=ModInRange_i(gActors[index].rgba.a,0,2);
+    gActors[index].ScaleX=ModInRange_i(gActors[index].ScaleX,-0.01f,gActors[index].unk_0x114);
+    gActors[index].ScaleY=ModInRange_i(gActors[index].ScaleY,-0.01f,gActors[index].unk_0x118);
+    if((gActors[index].rgba.a==0)||(gActors[index].ScaleX<0.0)||(gActors[index].ScaleY<0.0)){
+        gActors[index].flag=0;
+    }
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800348E4.s")
```

Comment:
immediate float rvalues are considered rodata as far as I know. Make sure argument types are correct, and make sure you have the rodata section correctly defined. Use NON_MATCHING ifdef gate. Consider spacing operators for better readability